src/App.jsx

```diff
@@ -1,21 +1,34 @@
 import "./App.css";
-import { useState } from "react";
+import React, { useState, useEffect } from "react";
 import TodoList from "./components/TodoList.jsx";
 import AddTodoForm from "./components/AddTodoForm.jsx";
 
 function App() {
-  const [todoList, setTodoList] = useState([]);
+  const useSemiPersistentState = (key, initialValue) => {
+    const [value, setValue] = useState(
+      JSON.parse(localStorage.getItem(key)) || initialValue
+    );
+    useEffect(() => {
+      localStorage.setItem(key, JSON.stringify(value));
+    }, [value, key]);
+    return [value, setValue];
+  };
+  const [todoList, setTodoList] = useSemiPersistentState(
+    "savedTodoList",
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
+  );
 
   const addTodo = (newTodo) => {
-    setTodoList([...todoList, newTodo]);
-  }
+    const newToDoList = [...todoList, newTodo];
+    setTodoList(newToDoList);
+  };
 
   return (
-    <div>
+    <React.Fragment>
```

Comment:
In my opinion, it’s better to import `Fragment` in line 2 ( `import React, { useState, useEffect, Fragment } from "react";`  ) and then refer to it as `Fragment`, or use the shorthand version `<></>`